### PR TITLE
Some minor layer housekeeping

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2950,9 +2950,8 @@ static VkBool32 validate_draw_state(layer_data *my_data, GLOBAL_CB_NODE *pCB, Vk
                 result |= validate_dynamic_offsets(my_data, pCB, activeSetNodes);
         }
         // Verify Vtx binding
-        if (pPipe->vtxBindingCount > 0) {
-            VkPipelineVertexInputStateCreateInfo *vtxInCI = &pPipe->vertexInputCI;
-            for (uint32_t i = 0; i < vtxInCI->vertexBindingDescriptionCount; i++) {
+        if (pPipe->vertexBindingDescriptions.size() > 0) {
+            for (uint32_t i = 0; i < pPipe->vertexBindingDescriptions.size(); i++) {
```
Comment (lentinem) [inline layers/core_validation.cpp]: Size mismatch for the comparison; should probably use size_t for i.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -3040,8 +3039,8 @@ static VkBool32 verifyPipelineCreateState(layer_data *my_data, const VkDevice de
 
     if (pPipeline->graphicsPipelineCI.pColorBlendState != NULL) {
         if (!my_data->physDevProperties.features.independentBlend) {
-            VkPipelineColorBlendAttachmentState *pAttachments = pPipeline->pAttachments;
-            for (uint32_t i = 1; i < pPipeline->attachmentCount; i++) {
+            VkPipelineColorBlendAttachmentState *pAttachments = &pPipeline->attachments[0];
+            for (uint32_t i = 1; i < pPipeline->attachments.size(); i++) {
```
Comment (lentinem) [inline layers/core_validation.cpp]: Ditto.

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -6532,9 +6521,6 @@ vkCreateGraphicsPipelines(VkDevice device, VkPipelineCache pipelineCache, uint32
         for (i = 0; i < count; i++) {
             if (pPipeNode[i]) {
```
Comment (tobine) [inline layers/core_validation.cpp]: We can kill this "if" check now, yes?

**Comment by chrisforbes** [comment]
Will fix this up and rebase it (there are some minor conflicts with the changes from #150) today.

File: layers/core_validation.cpp (hunk 3 above)
Comment (chrisforbes) [inline layers/core_validation.cpp]: Yes.

**Comment by chrisforbes** [comment]
Ready to go.

**Comment by lentinem** [comment]
+1

**Comment by tobine** [comment]
Merged to master

**Pull request closed without merging**